metadetect/lsst_metadetect.py

```diff
@@ -168,6 +177,10 @@ def _add_positions_and_psf(self, medsifier, res, shear_str, mbobs_list):
             else:
                 newres["mfrac"] = 0
 
+                # mark bad bboxes as masked
+                w, = np.where(newres['flags'] & procflags.BAD_BBOX != 0)
+                newres['mfrac'][w] = 1.0
+
```

Comment:
This block confuses me. Is mfrac == 1 even for bad boxes when we have an mfrac image with non-zero values?

Reply:
I think maybe this block should always be run (so move it out of the else block).

Reply:
well, mfrac is set to 1 in the mfrac calculating coded if the bbox is <= 0 so it would be redundant in that case

Reply:
I think I'll just remove this and people should use the damn flags

Reply:
Ahhhhh. We should not rely on this behavior unless we add a test for it for the mfrac code.

Reply:
uhhh.. no because that would be inconsistent with the case that the mfrac code gets run

Reply:
Can I push to this branch? I want to avoid a change in one spot depending on behavior in another. If we simply add a test to make sure the mfrac measurements are always 1 if bbox < 0 in the test suite, then the code as is is fine.

Reply:
go for it